src/main/java/com/jsoniter/IterImplForStreaming.java

```diff
@@ -487,48 +487,56 @@ public final static int readStringSlowPath(JsonIterator iter, int j) throws IOEx
         }
     }
 
-    static long readLongSlowPath(JsonIterator iter, long value) throws IOException {
+    static long readLongSlowPath(final JsonIterator iter, long value, final boolean negative) throws IOException {
```

Comment:
can we change `final boolean negative` to `final long limit`, so we do not need to `negative ? value : -value` in the end, as the place reading `-` already has this if statement in place.

Reply:
some additional param is required until we inline these slow path methods or split them on pair of methods (one for negative, other for positive)